packages/core/src/components/date-picker/date-picker.tsx

```diff
@@ -775,21 +786,27 @@ export class DatePicker {
               return (
                 <Fragment>
                   <div class="calendar-item week-number">{week.weekNumber}</div>
-                  {week.dayNumbers.map((day) => (
-                    <div
-                      key={day}
-                      id={`day-cell-${day}`}
-                      date-calender-day
-                      class={this.getDayClasses(day)}
-                      onClick={() => this.selectDay(day)}
-                      onKeyUp={(e) => e.key === 'Enter' && this.selectDay(day)}
-                      tabIndex={day === this.focusedDay ? 0 : -1}
-                      onFocus={() => this.onDayFocus()}
-                      onBlur={() => this.onDayBlur()}
-                    >
-                      {day}
-                    </div>
-                  ))}
+                  {week.dayNumbers.map((day) => {
+                    return day ? (
```

Comment:
Looks odd, why should the day numbers not be defined?

Reply:
Because the calendar will be padded with [undefined](https://github.com/siemens/ix/blob/main/packages/core/src/components/date-picker/date-picker.tsx#L432)s